protocols/v2/subprotocols/common-messages/src/setup_connection.rs

```diff
@@ -84,8 +84,23 @@ impl<'decoder> SetupConnection<'decoder> {
 
                 work_selection && version_rolling
             }
-            // TODO
-            _ => todo!(),
+            Protocol::JobDeclarationProtocol => {
+                let available = available_flags.reverse_bits();
+                let required = required_flags.reverse_bits();
+
+                let requires_async_job_mining_passed = (required >> 31) & 1 > 0;
+                let requires_async_job_mining_self = (available >> 31) & 1 > 0;
+
+                let specific_flags_check =
+                    !requires_async_job_mining_self || requires_async_job_mining_passed;
+                let general_flags_check = (available & required) == required;
+
+                specific_flags_check && general_flags_check
+            }
+            Protocol::TemplateDistributionProtocol | Protocol::JobDistributionProtocol => {
+                // Assuming these protocols do not define flags
```

Comment:
@Fi3 can you confirm Template Distribution Protocol does not set any flags for `SetupConnection`?

I cannot find anything on [the specs](https://stratumprotocol.org/specification/07-Template-Distribution-Protocol/) but just to make sure.

Reply:
I confirm

Reply:
ok so @Shourya742 we should change this comment

instead of saying we are assuming, we should confidently state this scenario does not have any meaningful flags for `SetupConnection` message